driver: phy: phy_qualcomm_ar8031: error: 'speeds' undeclared

### Describe the bug

- platform:
   imx8mn_evk/mimx8mn6/a53/smp and all other imx8mx platforms
- git bisect
847be49dbd737f45f1249e6ceac5d89921764166 
```
commit 847be49dbd737f45f1249e6ceac5d89921764166
Author: Fin Maaß <f.maass@vogl-electronic.com>
Date:   Mon Jun 16 12:58:29 2025 +0200

    drivers: ethernet: phy: add common functions for PHYs
    
    This commit adds common functions for PHYs to the mii driver.
    
    Signed-off-by: Fin Maaß <f.maass@vogl-electronic.com>

 drivers/ethernet/phy/phy_microchip_ksz8081.c | 39 ++------------
 drivers/ethernet/phy/phy_mii.c               | 79 +++++-----------------------
 drivers/ethernet/phy/phy_mii.h               | 63 ++++++++++++++++++++++
 drivers/ethernet/phy/phy_qualcomm_ar8031.c   | 54 +++----------------
 drivers/ethernet/phy/phy_realtek_rtl8211f.c  | 61 +++------------------
 drivers/ethernet/phy/phy_ti_dp83825.c        | 42 ++-------------
 drivers/ethernet/phy/phy_ti_dp83867.c        | 61 ++++-----------------
 7 files changed, 110 insertions(+), 289 deletions(-)
 create mode 100644 drivers/ethernet/phy/phy_mii.h
```

### Regression

- [x] This is a regression.

### Steps to reproduce

```
west build -p -b imx8mn_evk/mimx8mn6/a53/smp samples/posix/gettimeofday -T sample.posix.gettimeofday
```

### Relevant log output

> 
[190/259] Linking C static library zephyr/drivers/interrupt_controller/libdrivers__interrupt_controller.a[191/259] Building C object zephyr/drivers/clock_control/CMakeFiles/drivers__clock_control.dir/clock_control_mcux_ccm.c.obj[192/259] Linking C static library zephyr/drivers/clock_control/libdrivers__clock_control.a[193/259] Building C object zephyr/drivers/console/CMakeFiles/drivers__console.dir/uart_console.c.obj[194/259] Linking C static library zephyr/drivers/console/libdrivers__console.a[195/259] Building C object zephyr/drivers/ethernet/CMakeFiles/drivers__ethernet.dir/eth_nxp_enet.c.obj[196/259] Building C object zephyr/drivers/ethernet/CMakeFiles/drivers__ethernet.dir/phy/phy_qualcomm_ar8031.c.objFAILED: zephyr/drivers/ethernet/CMakeFiles/drivers__ethernet.dir/phy/phy_qualcomm_ar8031.c.obj on-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -fno-asynchronous-unwind-tables -ftls-model=local-exec -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=/home/jenkins/agent/workspace/zephyr_downstream_build_daily/zephyr/samples/posix/gettimeofday=CMAKE_SOURCE_DIR -fmacro-prefix-map=/home/jenkins/agent/workspace/zephyr_downstream_build_daily/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/home/jenkins/agent/workspace/zephyr_downstream_build_daily=WEST_TOPDIR -ffunction-sections -fdata-sections -moverride=tune=no_ldp_stp_qregs -specs=picolibc.specs -D_POSIX_THREADS -std=c99 -MD -MT zephyr/drivers/ethernet/CMakeFiles/drivers__ethernet.dir/phy/phy_qualcomm_ar8031.c.obj -MF zephyr/drivers/ethernet/CMakeFiles/drivers__ethernet.dir/phy/phy_qualcomm_ar8031.c.obj.d -o zephyr/drivers/ethernet/CMakeFiles/drivers__ethernet.dir/phy/phy_qualcomm_ar8031.c.obj -c /home/jenkins/agent/workspace/zephyr_downstream_build_daily/zephyr/drivers/ethernet/phy/phy_qualcomm_ar8031.c/home/jenkins/agent/workspace/zephyr_downstream_build_daily/zephyr/drivers/ethernet/phy/phy_qualcomm_ar8031.c: In function 'qc_ar8031_cfg_link':/home/jenkins/agent/workspace/zephyr_downstream_build_daily/zephyr/drivers/ethernet/phy/phy_qualcomm_ar8031.c:272:9: error: 'ret' undeclared (first use in this function) 272 
ret = phy_mii_set_anar_reg(dev, speeds);  ^~~/home/Jenkins/agent/workspace/zephyr_downstream_build_daily/zephyr/drivers/ethernet/phy/phy_qualcomm_ar8031.c:272:9: note: each undeclared identifier is reported only once for each function it appears in/home/jenkins/agent/workspace/zephyr_downstream_build_daily/zephyr/drivers/ethernet/phy/phy_qualcomm_ar8031.c:272:41: error: 'speeds' undeclared (first use in this function); did you mean 'adv_speeds'? 272 
ret = phy_mii_set_anar_reg(dev, speeds); 
 ^~~~~~ | adv_speedsninja: build stopped: subcommand failed.


### Impact

Annoyance – Minor irritation; no significant impact on usability or functionality.

### Environment

- OS: Linux
- toolchain: 0.17.0
- version: b9bfff29f4bf99ecb19dd2078bd06447e279cfcf

### Additional Context

_No response_

--- a/drivers/ethernet/phy/phy_qualcomm_ar8031.c
+++ b/drivers/ethernet/phy/phy_qualcomm_ar8031.c
@@ -255,10 +255,11 @@ static void monitor_work_handler(struct k_work *work)
 	k_work_reschedule(&data->monitor_work, K_MSEC(CONFIG_PHY_MONITOR_PERIOD));
 }
 
-static int qc_ar8031_cfg_link(const struct device *dev, enum phy_link_speed adv_speeds,
+static int qc_ar8031_cfg_link(const struct device *dev, enum phy_link_speed speeds,
 			      enum phy_cfg_link_flag flags)
 {
 	uint32_t bmcr_reg;
+	int ret;
 
 	if (flags & PHY_FLAG_AUTO_NEGOTIATION_DISABLED) {
 		LOG_ERR("Disabling auto-negotiation is not supported by this driver");

